src/speedrun.c

```diff
@@ -235,3 +239,96 @@ void speedrun_map_completed()
     // closes the connection
     mysql_close(con);
 }
+
+
+struct string {
+  char *ptr;
+  size_t len;
+};
+
+void init_string(struct string *s) {
+  s->len = 0;
+  s->ptr = malloc(s->len+1);
+  if (s->ptr == NULL) {
+    fprintf(stderr, "malloc() failed\n");
+    exit(EXIT_FAILURE);
+  }
+  s->ptr[0] = '\0';
+}
+
+size_t url_result_write(void *ptr, size_t size, size_t nmemb, struct string *s)
```

Comment:
Name may be changed, this function does nothing that is specific to url result.
It could be named, write_to_string or something